RecoTracker/LST/plugins/alpaka/LSTProducer.cc

```diff
@@ -0,0 +1,100 @@
+#include <alpaka/alpaka.hpp>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/SynchronizingEDProducer.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+#include "RecoTracker/LST/interface/LSTOutput.h"
+#include "RecoTracker/LST/interface/LSTPhase2OTHitsInput.h"
+#include "RecoTracker/LST/interface/LSTPixelSeedInput.h"
+
+#include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
+
+#include "RecoTracker/LSTCore/interface/alpaka/LST.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class LSTProducer : public stream::SynchronizingEDProducer<> {
```

Comment:
**question** what is the advantage of having separate producers that prepare the inputs used by the `LSTProducer` ?

Reply:
modularity? as a general principle.
I think the structure was following what's done for mkFit.

Reply:
I see.
I'm not sure whether that's a better approach than following the other 200+ reco packages in CMSSW, though.

Reply:
Anyway, what I meant is if there is any concrete use caseforeseen, _e.g._ building the input collection once and feeding it to two or more LST modules with different configurations ?

Reply:
> I see. I'm not sure whether that's a better approach than following the other 200+ reco packages in CMSSW, though.

that setup for mkFit was established by @makortel 
Perhaps he can comment.

Reply:
> Anyway, what I meant is if there is any concrete use case foreseen, _e.g._ building the input collection once and feeding it to two or more LST modules with different configurations ?

it was explored already in the HLT context last year. It may be revisited again.

Reply:
> > I see. I'm not sure whether that's a better approach than following the other 200+ reco packages in CMSSW, though.
> 
> that setup for mkFit was established by @makortel Perhaps he can comment.

I don't remember anymore the exact details. I can think of
* One collection of hits serves multiple instances of `MkFitProducer`
* Support for phase2 input hits was possible with a separate EDProducer rather than adding more conditional code (or a helper plugin factory) in `MkFitProducer`
* IIRC with both hits and seeds there was an idea to eventually convert the present `TrackingRecHitSoA` and `TrackSoA` directly to mkFit input types avoiding the conversion to legacy hit and seed objects
* Maybe with the mkFit output there was hope to eventually have (at least some) downstream to consume directly the mkFit Track collection (and/or develop the mkFit Track collection towards being more generally useful)